cdap-data-fabric/src/test/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetServiceAuthorizationTest.java

```diff
@@ -174,15 +181,34 @@ public void execute() throws Exception {
     SecurityRequestContext.setUserId(BOB.getName());
     Assert.assertEquals(ImmutableSet.of(dsId2),
                         summaryToDatasetIdSet(dsFramework.getInstances(NamespaceId.DEFAULT)));
-    deleteAndAssertPrivileges(dsId2);
+    dsFramework.deleteInstance(dsId2);
   }
 
   @Test
   public void testNotFound() throws Exception {
-    String namespace = NamespaceId.DEFAULT.getNamespace();
     final DatasetId nonExistingInstance = NamespaceId.DEFAULT.dataset("notfound");
     final DatasetModuleId nonExistingModule = NamespaceId.DEFAULT.datasetModule("notfound");
     final DatasetTypeId nonExistingType = NamespaceId.DEFAULT.datasetType("notfound");
+    try {
+      // user will not be able to get the info about the instance since he does not have any privilege on the instance
+      dsFramework.getDatasetSpec(nonExistingInstance);
+      Assert.fail();
+    } catch (Exception e) {
```

Comment:
This will always throw `DatasetManagementException`, will check the message to make sure the test runs correctly